includes/chart.js

```diff
@@ -0,0 +1,330 @@
+                newNode = document.createElementNS("http://www.w3.org/2000/svg", "ellipse");
+                newNode.setAttribute("rx", template_node.radiusX); // Use the horizontal radius attribute
+                newNode.setAttribute("ry", template_node.radiusY); // Use the vertical radius attribute
+                newNode.setAttribute("cx", x);
+                newNode.setAttribute("cy", y);
+            } else if (template_node.shape === 'rect') {
+                // Create a rectangle for the node
+                newNode = document.createElementNS("http://www.w3.org/2000/svg", "rect");
+                newNode.setAttribute("x", x);
+                newNode.setAttribute("y", y);
+                newNode.setAttribute("width", template_node.width); // Use the width attribute
+                newNode.setAttribute("height", template_node.height); // Use the height attribute
+                if (template_node.rx && template_node.ry) {
+                    newNode.setAttribute("rx", template_node.rx); // Set horizontal radius for rounded corners
+                    newNode.setAttribute("ry", template_node.ry); // Set vertical radius for rounded corners
+                }
+            }
+
+            newNode.setAttribute("id", `${row.id}_${template_node.id}`);
+            newNode.setAttribute("fill", colors[row.statementType]);
+            newNode.setAttribute("stroke", "black")
+
+            // Add text inside the shape
+            var text = document.createElementNS("http://www.w3.org/2000/svg", "text");
+            text.textContent = row[template_node.type];
+            text.setAttribute("fill", "white");
+            text.setAttribute("font-size", "12px");
+            text.setAttribute("font-family", "calibri");
+            text.setAttribute("text-anchor", "middle"); // Center align the text
+
+            var textX = x, textY = y;
+            if (template_node.shape === 'rect') {
+                if (template_node.rx && template_node.ry) {
+                    // Calculate center position for rectangles with rounded corners
+                    textX += template_node.rx + (template_node.width - 2 * template_node.rx) / 2;
+                    textY += template_node.ry + (template_node.height - 2 * template_node.ry) / 2;
+                } else {
+                    // Calculate center position for rectangles without rounded corners
+                    textX += template_node.width / 2;
+                    textY += template_node.height / 2;
+                }
+            }
+            text.setAttribute("x", textX);
+            text.setAttribute("y", textY);
+
+            group.appendChild(newNode);
+            group.appendChild(text);
+            svg.appendChild(group);
+
+            // Add click event to display the clicked node
+            newNode.addEventListener('click', function () {
+                document.getElementById("clicked-node").textContent = "Clicked Node: " + `${row.id}_${template_node.id}`;
```

Comment:
The ID "clicked-node" could maybe named after it's function/role, not after what happens if you do something outside of this `div`. Maybe "node-logging"? I don't know enough about its function to come up with a good one. :-)

Reply:
This specific `clicked-node` div is a development leftover and currently serves as a placeholder for eventually highlighting the relevant row in the table when a node/statement gets selected.